lib/rucio/daemons/oauthmanager/oauthmanager.py

```diff
@@ -38,140 +38,139 @@
+
+    except (DatabaseException, DatabaseError) as err:
+        if match('.*QueuePool.*', str(err.args[0])):
+            logger(logging.WARNING, traceback.format_exc())
+            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
+        elif match('.*ORA-03135.*', str(err.args[0])):
+            logger(logging.WARNING, traceback.format_exc())
+            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
+        else:
             logger(logging.CRITICAL, traceback.format_exc())
             record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
 
-        tottime = time.time() - start
-        logger(logging.INFO, 'took %f seconds to delete %i tokens, %i session parameters and refreshed %i tokens', tottime, ndeleted, ndeletedreq, nrefreshed)
-        record_timer(name='oauth_manager.duration', time=1000 * tottime)
-
-        if once:
-            break
+    try:
+        # make a heartbeat
+        worker_number, total_workers, logger = heartbeat_handler.live()
+
+        # DELETING EXPIRED OAUTH SESSION PARAMETERS
+        logger(logging.INFO, '----- START ----- DELETION OF EXPIRED OAUTH SESSION REQUESTS ----- ')
+        logger(logging.INFO, 'starting deletion of expired OAuth session requests')
+        ndeletedreq += delete_expired_oauthrequests(total_workers, worker_number, limit=max_rows)
+        logger(logging.INFO, 'expired parameters of %i authentication requests were deleted', ndeletedreq)
+        logger(logging.INFO, '----- END ----- DELETION OF EXPIRED OAUTH SESSION REQUESTS ----- ')
+        record_counter(name='oauth_manager.oauthreq.deleted', delta=ndeletedreq)
+
+    except (DatabaseException, DatabaseError) as err:
+        if match('.*QueuePool.*', str(err.args[0])):
+            logger(logging.WARNING, traceback.format_exc())
+            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
+        elif match('.*ORA-03135.*', str(err.args[0])):
+            logger(logging.WARNING, traceback.format_exc())
+            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
         else:
-            daemon_sleep(start_time=start, sleep_time=sleep_time, graceful_stop=GRACEFUL_STOP)
+            logger(logging.CRITICAL, traceback.format_exc())
+            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
 
-    die(executable=executable, hostname=socket.gethostname(), pid=os.getpid(), thread=threading.current_thread())
-    logger(logging.INFO, 'graceful stop done')
+    tottime = time.time() - start
+    logger(logging.INFO, 'took %f seconds to delete %i tokens, %i session parameters and refreshed %i tokens', tottime, ndeleted, ndeletedreq, nrefreshed)
+    record_timer(name='oauth_manager.duration', time=1000 * tottime)
```

Comment:
record_timer doesn't exist anymore. Please rebase the PR to use the changed behavior from https://github.com/rucio/rucio/pull/5718.